crates/common/src/provider/runtime_transport.rs

```diff
@@ -76,6 +86,18 @@ pub struct RuntimeTransport {
     jwt: Option<String>,
     /// The timeout for requests.
     timeout: std::time::Duration,
+    /// The current amount of requests being retried.
+    requests_enqueued: Arc<AtomicU32>,
+    /// The policy that will be used to determine whether to retry a request or not.
+    policy: RateLimitRetryPolicy,
+    /// Max amount of retries.
+    max_rate_limit_retries: u32,
+    /// Max amount of timeout retries
+    max_timeout_retries: u32,
+    /// Initial backoff
+    initial_backoff: u64,
+    /// Compute units per second
+    compute_units_per_second: u64,
```

Comment:
Discussed offline, this will eventually become a service reusable in alloy, but we need further changes in Alloy before we can move forward to making this a service